src/main/java/com/codetaylor/mc/onslaught/modules/onslaught/event/handler/InvasionKillCountUpdateEventHandler.java

```diff
@@ -44,6 +43,41 @@ public void on(LivingDeathEvent event) {
     // Checked above
     //noinspection ConstantConditions
     PlayerList playerList = minecraftServer.getPlayerList();
-    this.invasionKillCountUpdater.onDeath(invasionGlobalSavedData, entityData, playerList::getPlayerByUUID);
+    this.invasionKillCountUpdater
+        .onDeath(invasionGlobalSavedData, entityData, playerList::getPlayerByUUID);
+  }
+
+  /**
+   * @param event An explosion event, so we count suiciding creepers
+   */
+  @SubscribeEvent
+  public void on(ExplosionEvent.Start event) {
+    EntityLivingBase entity = event.getExplosion().getExplosivePlacedBy();
+    if (entity == null) {
+      return;
+    }
+
+    World world = entity.world;
+
+    if (world.isRemote ||
+        !(world instanceof WorldServer)) {
+      return;
+    }
+
+    // we know we're in a remote world. Pretty sure there's a cleaner way, but bug fix first.
+    // noinspection ConstantConditions
+    this.invasionKillCountUpdater.onDeath(
+        InvasionGlobalSavedData
+            .get(world),
+        entity
+            .getEntityData(),
+        world
+            .getMinecraftServer()
+            .getPlayerList()::getPlayerByUUID
+    );
+  }
+
+  public void on(Event event) {
```

Comment:
Why is this extra `on` method necessary?

Reply:
Because I got distracted and left it dangling. Removed.